### PR TITLE
Fix structure of Kinesis TOC

### PR DESCRIPTION
**Pull request by dedemorton**
Summary of changes:
* Fixes the structure of the Kinesis TOC as described in #2893.
* Organizes all the Kinesis-related content into a single folder to make it easier to move the content in the future.
* Adds short titles to avoid repetition in the TOC.

This PR must be merged at the same time as https://github.com/elastic/docs/pull/2674

TOC before:
image

TOC after:
image

### CODE REVIEW

**Comment by dedemorton** [comment]
@tommyers-elastic FYI

File: docs/en/aws-firehose/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+:doctype: book
+
+= Amazon Kinesis Data Firehouse Ingestion Guide
```
Comment (dedemorton) [inline docs/en/aws-firehose/index.asciidoc:3]:
```suggestion
= Amazon Kinesis Data Firehose Ingestion Guide
```

File: docs/en/aws-firehose/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+:doctype: book
+
+= Amazon Kinesis Data Firehose Ingestion Guide
```
Comment (dedemorton) [inline docs/en/aws-firehose/index.asciidoc:3]:
```suggestion
= Amazon Kinesis Data Firehose Ingest Guide
```

**Review by bmorelli25 — APPROVED** [review]
I didn't test this, but the changes make sense.

**Pull request merged**